examples/object_detection/android-jetpack-compose/app/src/main/java/com/example/objectdetection/home/gallery/GalleryView.kt

```diff
@@ -0,0 +1,138 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.objectdetection.ui.theme.Turquoise
+
+// Here we have the gallery view which is displayed in Home screen
+
+// It's used to run object detection on images and videos from the phone's storage
+
+// It takes as input the object detection options, and a function to update the inference time state
+@Composable
+fun GalleryView(
+    threshold: Float,
+    maxResults: Int,
+    delegate: Int,
+    mlModel: Int,
+    setInferenceTime: (Int) -> Unit,
+) {
+
+    // We need a state to hold the Uri of the currently chosen media (image or video) if any
+    var selectedMediaUri by rememberSaveable {
+        mutableStateOf<Uri?>(null)
+    }
+
+
+    // We use this launcher later to launch a new activity to select a media file from
+    val galleryLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.OpenDocument(),
+        // On selecting an image or a video, we update the selected media uri
+        onResult = { uri ->
+            selectedMediaUri = uri
+        },
+    )
+
+    Box(modifier = Modifier.fillMaxSize()) {
+        // We first want to figure out the type of the selected media, which is one of
+        // the following three cases: Image, Video, Unknown.
+
+        val selectedMediaType = selectedMediaUri?.let {
+            val mimeType = LocalContext.current.contentResolver.getType(it)
+            if (mimeType == null) {
+                "Unknown"
+            } else if (mimeType.startsWith("image")) {
+                "Image"
+            } else if (mimeType.startsWith("video")){
+                "Video"
+            } else {
+                "Unknown"
+            }
+        }
+
+        // Now that we know the selected media type, we display the appropriate composable
+
+        when (selectedMediaType) {
+            "Image" -> ImageDetectionView(
+                threshold = threshold,
+                maxResults = maxResults,
+                delegate = delegate,
+                mlModel = mlModel,
+                setInferenceTime = setInferenceTime,
+                imageUri = selectedMediaUri!!,
+            )
+            "Video" -> VideoDetectionView(
+                threshold = threshold,
+                maxResults = maxResults,
+                delegate = delegate,
+                mlModel = mlModel,
+                setInferenceTime = setInferenceTime,
+                videoUri = selectedMediaUri!!,
+            )
+            "Unknown" -> Text(
+                text = "Unknown media type",
+                modifier = Modifier.align(
+                    Alignment.Center
+                ),
+            )
+            // In this case, the selected media uri is null, i.e, nothing is selected
+            else -> Text(
+                text = "Click + to add an image or a video to begin running the object detection.",
```

Comment:
String resource

Reply:
Again, I'll leave that part as we discussed previously